hivemq-getting-started-cloud/src/main/java/com/hivemq/CloudExample.java

```diff
@@ -0,0 +1,84 @@
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+public class CloudExample {
+
+    public static void main(String[] args) throws InterruptedException, UnknownHostException, SocketException {
+        final String host = "<your_host>"; // use your host-name, it should look like '<alphanumeric>.s2.eu.hivemq.cloud'
+        final String username = "<your_username>"; // your credentials
+        final String password = "<your_password>";
+
+        InetAddress localHost = InetAddress.getLocalHost();
+        NetworkInterface ni = NetworkInterface.getByInetAddress(localHost);
+        byte[] hardwareAddress = ni.getHardwareAddress(); // use this to get your MAC address, use it as a unique identifier
+
+        // 1. create the client
+        final Mqtt5Client client = Mqtt5Client.builder()
+                .identifier("sensor-" + hardwareAddress) // use a unique identifier
+                .serverHost(host)
+                .automaticReconnectWithDefaultConfig() // the client automatically reconnects
+                .serverPort(8883) // this is the port of your cluster, for mqtt it is the default port 8883
+                .sslWithDefaultConfig() // establish a secured connection to HiveMQ Cloud using TLS
+                .build();
+
+
+        // 2. connect the client
+        client.toBlocking().connectWith()
+                .simpleAuth() // using authentication, which is required for a secure connection
+                .username(username) // use the username and password you just created
+                .password(password.getBytes(StandardCharsets.UTF_8))
+                .applySimpleAuth()
+                .willPublish() // the last message, before the client disconnects
+                .topic("home/will")
+                .payload("sensor gone".getBytes())
+                .applyWillPublish()
+                .send();
+
+        // 3. subscribe and consume messages
+        client.toAsync().subscribeWith()
+                .topicFilter("home/#")
+                .callback(publish -> {
+                    System.out.println("Received message on topic " + publish.getTopic() + ": " +
+                            new String(publish.getPayloadAsBytes(), StandardCharsets.UTF_8));
+                })
+                .send();
+
+        // 4. simulate periodic publishing of sensor data
+        while (true) {
+            client.toBlocking().publishWith()
+                    .topic("home/brightness")
+                    .payload(getBrightness())
+                    .send();
+
+            TimeUnit.MILLISECONDS.sleep(500);
+
+            client.toBlocking().publishWith()
+                    .topic("home/temperature")
+                    .payload(getTemperature())
+                    .send();
+
+            TimeUnit.MILLISECONDS.sleep(500);
+        }
```

Comment:
This can be done way prettier with the Rx flavour of the MqttClient. 

Doing blocking stuff in a while true loop is not really that good.